pkg/discovery/network/openshift4.go

```diff
@@ -90,13 +95,18 @@ func parseOS4Network(cr *unstructured.Unstructured) (*ClusterNetwork, error) {
 		result.ServiceCIDRs = append(result.ServiceCIDRs, serviceNetwork.(string))
 	}
 
-	result.NetworkPlugin, found, err = unstructured.NestedString(cr.Object, "spec", "networkType")
-
+	ocpNetworkType, found, err := unstructured.NestedString(cr.Object, "spec", "networkType")
 	if err != nil {
 		return nil, errors.Wrap(err, "error retrieving spec.networkType field")
 	} else if !found {
 		return nil, fmt.Errorf("field .spec.networkType expected, but not found in Network resource: %v", cr.Object)
 	}
 
+	if ocpNetworkType == NetworkTypeCalico {
+		result.NetworkPlugin = constants.NetworkPluginCalico
```

Comment:
Instead of importing this constant from `submariner` repo, we can create a new constant `CalicoNetworkPlugin` in `internal/constant/constants.go`

Reply:
We use the submariner-imported network plugins constants for all plugins discovery code in the operator, I adopted the same approach also for CalicoNetworkPlugin.

If we want to maintain a separate plugins constant list in the operator's repo, it makes sense to do that for all plugins and not just CalicoNetworkPlugin.

IMO, we should maintain the constants in one place in order to avoid confusion and inconsistency. So, the current implementation seems right to me.

Please, let me know if you still think the code needs to be changed

Reply:
Lets keep it as is then. Thanks for the explanation.